views/default/forms/useradd.php

```diff
@@ -23,62 +25,58 @@
 	$admin = array_shift($admin);
 }
 
-?>
-<div>
-	<label><?php echo elgg_echo('name');?></label><br />
-	<?php
-	echo elgg_view('input/text', array(
-		'name' => 'name',
-		'value' => $name,
-	));
-	?>
-</div>
-<div>
-	<label><?php echo elgg_echo('username'); ?></label><br />
-	<?php
-	echo elgg_view('input/text', array(
-		'name' => 'username',
-		'value' => $username,
-	));
-	?>
-</div>
-<div>
-	<label><?php echo elgg_echo('email'); ?></label><br />
-	<?php
-	echo elgg_view('input/text', array(
-		'name' => 'email',
-		'value' => $email,
-	));
-	?>
-</div>
-<div>
-	<label><?php echo elgg_echo('password'); ?></label><br />
-	<?php
-	echo elgg_view('input/password', array(
-		'name' => 'password',
-		'value' => $password,
-	));
-	?>
-</div>
-<div>
-	<label><?php echo elgg_echo('passwordagain'); ?></label><br />
-	<?php
-	echo elgg_view('input/password', array(
-		'name' => 'password2',
-		'value' => $password2,
-	));
-	?>
-</div>
-<div>
-<?php 
-	echo elgg_view('input/checkboxes', array(
-		'name' => "admin",
-		'options' => array(elgg_echo('admin_option') => 1),
-		'value' => $admin,
-	));
-?>
-</div>
+echo elgg_view_input('text', [
+	'name' => 'name',
+	'value' => $name,
+	'label' => elgg_echo('name'),
+	'required' => true,
+]);
+
+echo elgg_view_input('text', [
+	'name' => 'username',
+	'value' => $username,
+	'label' => elgg_echo('username'),
+	'required' => true,
+]);
+
+echo elgg_view_input('email', [
+	'name' => 'email',
+	'value' => $email,
+	'label' => elgg_echo('email'),
+	'required' => true,
+]);
+
+echo elgg_view_input('checkbox', array(
+	'name' => 'autogen_password',
+	'value' => 1,
+	'default' => false,
+	'label' => elgg_echo('autogen_password_option'),
+	'checked' => false, // not using sticky value to make JS implementation cleaner
```

Comment:
I mean if we add hidden class right away, we need to add boilerplate JS for
toggling classes/required attributes etc
On Jun 13, 2016 6:28 PM, "Steve Clay" <notifications@github.com> wrote:

> In views/default/forms/useradd.php
> <https://github.com/Elgg/Elgg/pull/9892#discussion_r66823570>:
>
> > +	'required' => true,
> > +]);
> > +
> > +echo elgg_view_input('email', [
> > +	'name' => 'email',
> > +	'value' => $email,
> > +	'label' => elgg_echo('email'),
> > +	'required' => true,
> > +]);
> > +
> > +echo elgg_view_input('checkbox', array(
> > +	'name' => 'autogen_password',
> > +	'value' => 1,
> > +	'default' => false,
> > +	'label' => elgg_echo('autogen_password_option'),
> > +	'checked' => false, // not using sticky value to make JS implementation cleaner
>
> Actually you don't need to set default, because the hidden input can't
> emit a change event anyway
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9892/files/e155ca741eea6ed4f4034b405c617ef9a6647161#r66823570>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABJaSXLHbZent3KVr4t9sJFSnjUDFd-sks5qLYU2gaJpZM4I0Q3X>
> .
>